src/owr/OptimisticWithdrawalRecipientPectra.sol

```diff
@@ -0,0 +1,451 @@
+    return _getArgAddress(RECOVERY_ADDRESS_OFFSET);
+  }
+
+  /// Get OWR tranche `i`
+  /// @dev emulates to uint256[] internal immutable tranche;
+  function _getTranche(uint256 i) internal pure returns (uint256) {
+    unchecked {
+      // shouldn't overflow
+      return _getArgUint256(TRANCHES_OFFSET + i * ONE_WORD);
+    }
+  }
+
+  /// -----------------------------------------------------------------------
+  /// storage - mutables
+  /// -----------------------------------------------------------------------
+  /// @dev set to `true` after owner is initialized
+  bool public initialized;
+
+  /// Amount of active balance set aside for pulls
+  /// @dev ERC20s with very large decimals may overflow & cause issues
+  uint128 public fundsPendingWithdrawal;
+
+  /// Amount of distributed OWRecipient token for principal
+  /// @dev Would be less than or equal to amount of stake
+  /// @dev ERC20s with very large decimals may overflow & cause issues
+  uint128 public claimedPrincipalFunds;
+
+  /// Mapping to account balances for pulling
+  mapping(address => uint256) internal pullBalances;
+
+  /// -----------------------------------------------------------------------
+  /// constructor
+  /// -----------------------------------------------------------------------
+
+  // solhint-disable-next-line no-empty-blocks
+  /// clone implementation doesn't use constructor
+  constructor(address _pectraWithdrawalAddress, address _pectraConsolidationAddress) {
+    pectraConsolidationAddress = _pectraConsolidationAddress;
+    pectraWithdrawalAddress = _pectraWithdrawalAddress;
+  }
+
+  /// -----------------------------------------------------------------------
+  /// functions
+  /// -----------------------------------------------------------------------
+
+  /// -----------------------------------------------------------------------
+  /// functions - public & external
+  /// -----------------------------------------------------------------------
+
+  /// @dev initializes the owner
+  /// @param _owner the owner address
+  function initialize(address _owner) public {
+    if (initialized) revert InvalidRequest_Initialize();
+    _initializeOwner(_owner);
+    initialized = true;
+  }
+
+  /// emit event when receiving ETH
+  /// @dev implemented w/i clone bytecode
+  /* receive() external payable { */
+  /*     emit ReceiveETH(msg.value); */
+  /* } */
+
+  function requestConsolidation(bytes calldata source, bytes calldata target)
+    external
+    payable
+    onlyOwnerOrRoles(CONSOLIDATION_WITHDRAWAL_ROLE)
+  {
+    if (source.length != 48 || target.length != 48) revert InvalidConsolidation_Failed();
+
+    // ;; Input data has the following layout:
+    // ;;
+    // ;;  +--------+--------+
+    // ;;  | source | target |
+    // ;;  +--------+--------+
+    // ;;      48       48
+
+    (bool ret,) = pectraConsolidationAddress.call{value: msg.value}(abi.encodePacked(source, target));
```

Comment:
This wasn't tackled yet. I will check this out

Reply:
Added an option for this. We don't have a clear example for how the fee will be computed in production but I followed the `eas` files of main and fake_expo to understand what they are doing. Did a similar behavior on our side in the last commit

Reply:
I'm not 100% sure about it yet. Also they don't mention about it, other than using the `fake_expo.eas` contract. Which I assume it's similar to a production ready fee computation contract. In case it follows the same interface, our part should work as well

Reply:
cc @OisinKyne 

Maybe @samparsky can you take a look at the last commit as well, please?